src/NzbDrone.Core/Download/Clients/Flood/Flood.cs

```diff
@@ -0,0 +1,214 @@
+
+            var list = _proxy.GetTorrents(Settings);
+
+            foreach (var torrent in list)
+            {
+                var properties = torrent.Value;
+
+                if (!Settings.Tags.All(tag => properties.Tags.Contains(tag)))
+                {
+                    continue;
+                }
+
+                var item = new DownloadClientItem
+                {
+                    DownloadClientInfo = DownloadClientItemClientInfo.FromDownloadClient(this),
+                    DownloadId = torrent.Key,
+                    Title = properties.Name,
+                    OutputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(properties.Directory)),
+                    Category = properties.Tags.Count > 0 ? properties.Tags[0] : null,
+                    RemainingSize = properties.SizeBytes - properties.BytesDone,
+                    RemainingTime = TimeSpan.FromSeconds(properties.Eta),
+                    TotalSize = properties.SizeBytes,
+                    SeedRatio = properties.Ratio,
+                    Message = properties.Message,
+                };
+
+                if (properties.Status.Contains("error"))
+                {
+                    item.Status = DownloadItemStatus.Warning;
+                }
+                else if (properties.Status.Contains("seeding") || properties.Status.Contains("complete"))
+                {
+                    item.Status = DownloadItemStatus.Completed;
+                }
+                else if (properties.Status.Contains("downloading"))
+                {
+                    item.Status = DownloadItemStatus.Downloading;
+                }
+                else if (properties.Status.Contains("stopped"))
+                {
+                    item.Status = DownloadItemStatus.Paused;
+                }
+
+                item.CanMoveFiles = item.CanBeRemoved = item.Status == DownloadItemStatus.Completed;
+
+                items.Add(item);
+            }
+
+            return items;
+        }
+
+        public override DownloadClientItem GetImportItem(DownloadClientItem item, DownloadClientItem previousImportAttempt)
+        {
+            var result = item.Clone();
+
+            var contentPaths = _proxy.GetTorrentContentPaths(item.DownloadId, Settings);
+
+            if (contentPaths.Count < 2)
+            {
+                // For single-file torrent, OutputPath should be the path of file.
+                result.OutputPath = item.OutputPath + new OsPath(contentPaths[0]);
+            }
+            else
+            {
+                // For multi-file torrent, OutputPath should be the path of base directroy of torrent.
+                // isBasePath is false by default. Presumably base directory would be first fragment of path of content.
+                result.OutputPath = item.OutputPath + new OsPath(contentPaths[0].Split(new char[] { '\\', '/' }, StringSplitOptions.RemoveEmptyEntries)[0]);
```

Comment:
This is not working for multi-file torrents. It currently returns the path to the first file, but instead I believe it should just be returning `item.OutputPath`.

Reply:
`item.OutputPath` is the `directory` of `TorrentProperties`, which, by definition, is always a directory. This directory may or may not be the base directory. This causes problems as Sonarr can grab contents of unrelated torrents which reside in the same destination directory.

This line of code tries to combine the directory with the first segment (directory) of a content when the torrent is multi-file:

TorrentProperties.directory + first path segment of first content of a multi-file torrent (path -> split -> pathSegments[0])

`/mnt/data0/downloads` + **`Big.Buck.Bunny.1080P`**/~Soundtracks/Bunny.wav~

I tested with a multi-file torrent. It seems that this approach works fine as torrent clients always create a sub-directory for multi-file torrents by default. 

For single-file torrents, Sonarr expects path to a single file, which is handled by another case (above). 

What's not working? Did this line of code fail to get the first segment, combine paths, or...?

Reply:
> It currently returns the path to the first file.

It gave me the path to a text file, not a folder. Even if it happened to give me a video file it'd be wrong for a multi-video file download.

![image](https://user-images.githubusercontent.com/413544/104548770-5e6d6a00-55e6-11eb-9550-36563763025b.png)
`result.OutputPath` is set to `C:\downloads\torrents\test\The.Rookie.S03E02.WEBRip.x264-ION10\RARBG.txt`.
`item.OutputPath` is set to `C:\timebomb\downloads\torrents\test\The.Rookie.S03E02.WEBRip.x264-ION10`

Reply:
In your case, `directory` is the base directory, which is indeed a possible case. This case is not unhandled here. I assume there are some differences in how clients handle root directory of a multi-file torrent. I will work out a more robust method for detection of base directory.

Reply:
Yup, they pretty much universally suck at it, we're sometimes left guessing which has lead to issues in the past and somewhat recently with qbit.

Reply:
It is done. Please try it.

Reply:
Sonarr can probably utilize a method that returns an array of paths (of contents of torrents) instead. That allows things to be much more precise and eliminates any interferences.

Reply:
Working much better now. In the future please don't squash fixes immediately, we can do that when merging the PR.

Reply:
Ack. It is a shame that Github PR does not have the capability to compare different versions of change sets directly. I typically use more professional Code Review tools like Gerrit, which does have the capability (and as result, pushing additional commits can be _"stupid"_).

Reply:
Fortunately, though, there is a commit compare function: https://github.com/jesec/Sonarr/compare/17237a0e..0cae35f3.